Content.Shared/Sericulture/SericultureSystem.cs

```diff
@@ -0,0 +1,81 @@
+using Content.Shared.Actions;
+using Content.Shared.Actions.ActionTypes;
+using Content.Shared.DoAfter;
+using Content.Shared.Nutrition.EntitySystems;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+using Content.Shared.Popups;
+
+namespace Content.Shared.Sericulture;
+
+/// <summary>
+/// Allows mobs to produce materials with <see cref="SericultureComponent"/>.
+/// </summary>
+public abstract partial class SharedSericultureSystem : EntitySystem
+{
+    [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doAfterSystem = default!;
+    [Dependency] private readonly IPrototypeManager _protoManager = default!;
+    [Dependency] private readonly HungerSystem _hungerSystem = default!;
+    [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<SericultureComponent, ComponentInit>(OnCompInit);
+        SubscribeLocalEvent<SericultureComponent, ComponentShutdown>(OnCompRemove);
+        SubscribeLocalEvent<SericultureComponent, SericultureActionEvent>(OnSericultureStart);
+    }
+
+    /// <summary>
+    /// Giveths the action to preform sericulture on the entity
+    /// </summary>
+    private void OnCompInit(EntityUid uid, SericultureComponent comp, ComponentInit args)
+    {
+        if (!_protoManager.TryIndex<InstantActionPrototype>(comp.ActionProto, out var actionProto))
```

Comment:
if this is properly predicted the action prototype will need to be changed from using `serverEvent` im pretty sure

Reply:
you didnt change it in this PR so i cant github review it but im talking about https://github.com/PixelTheKermit/space-station-14/blob/f189b73fe57d9eaa72f043f2fca49754e7d161b8/Resources/Prototypes/Actions/spider.yml#L14

Reply:
Right think I got it, thanks!